### PR TITLE
[TT-10815] Implement OAS api level request header transformation

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Parent: https://tyktech.atlassian.net/browse/TT-4851
Subtask: https://tyktech.atlassian.net/browse/TT-10815

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7293996604)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7293996641)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7293996641)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7293996641)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7293996641)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit: db14499cadc8636021eb025d5dfb00711c720fb4 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7294130889)

File: apidef/migration.go [hunk 1]
```diff
@@ -311,6 +313,14 @@ func (a *APIDefinition) migrateCustomDomain() {
 	}
 }
 
+func (a *APIDefinition) migrateGlobalHeaders() {
+	vInfo := a.VersionData.Versions[""]
```
Comment (titpetric) [inline apidef/migration.go:317]: Can we have some sort of const for "" so there's a comment explaining the key?

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5904/merge
Commit: c8e75a0e4d3c975dbd917ad9e6341b726ffc8d28 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7300748123)

**Review by titpetric — APPROVED** [review]

**Pull request merged**